deployment/ccip/changeset/cs_ccip_home.go

```diff
@@ -35,29 +36,127 @@ import (
+	if c.CommitOffChainConfig == nil && c.ExecuteOffChainConfig == nil {
+		return errors.New("at least one of CommitOffChainConfig or ExecuteOffChainConfig must be set")
 	}
-	if err := c.ExecuteOffChainConfig.Validate(); err != nil {
-		return fmt.Errorf("invalid execute off-chain config: %w", err)
+	if c.CommitOffChainConfig != nil {
+		if err := c.CommitOffChainConfig.Validate(); err != nil {
+			return fmt.Errorf("invalid commit off-chain config: %w", err)
+		}
+		for tokenAddr, tokenConfig := range c.CommitOffChainConfig.TokenInfo {
+			tokenUnknownAddr, err := ccipocr3.NewUnknownAddressFromHex(string(tokenAddr))
+			if err != nil {
+				return fmt.Errorf("invalid token address %s: %w", tokenAddr, err)
+			}
+			aggregatorUnknownAddr, err := ccipocr3.NewUnknownAddressFromHex(string(tokenConfig.AggregatorAddress))
+			if err != nil {
+				return fmt.Errorf("invalid aggregator address %s: %w", tokenConfig.AggregatorAddress, err)
+			}
+			aggregatorAddr := common.HexToAddress(aggregatorUnknownAddr.String())
+			token := common.HexToAddress(tokenUnknownAddr.String())
+			tokenInfos := make([]tokenInfo, 0)
+			onchainState, ok := state.Chains[selector]
+			if !ok {
+				return fmt.Errorf("chain %d does not exist in state", selector)
+			}
+			for _, tk := range onchainState.BurnMintTokens677 {
+				tokenInfos = append(tokenInfos, tk)
+			}
+			for _, tk := range onchainState.ERC20Tokens {
+				tokenInfos = append(tokenInfos, tk)
+			}
+			for _, tk := range onchainState.ERC677Tokens {
+				tokenInfos = append(tokenInfos, tk)
+			}
+			tokenInfos = append(tokenInfos, onchainState.LinkToken)
+			tokenInfos = append(tokenInfos, onchainState.Weth9)
+			symbol, decimal, err := findTokenInfo(tokenInfos, token)
+			if err != nil {
+				return err
+			}
+			if decimal != tokenConfig.Decimals {
+				return fmt.Errorf("token %s -address %s has %d decimals in provided token config, expected %d",
+					symbol, token.String(), tokenConfig.Decimals, decimal)
+			}
+			feedChainState := state.Chains[feedChainSel]
+			aggregatorInState := feedChainState.USDFeeds[TokenSymbol(symbol)]
+			if aggregatorInState == nil {
+				if aggregatorAddr != (common.Address{}) {
+					return fmt.Errorf("token %s -address %s has no aggregator in state,"+
+						" but the aggregator %s is provided in token config", symbol, token.String(), aggregatorAddr.String())
+				}
+			} else {
+				if aggregatorAddr != aggregatorInState.Address() {
+					return fmt.Errorf("token %s -address %s has aggregator %s in provided token config, expected %s",
+						symbol, token.String(), aggregatorAddr.String(), aggregatorInState.Address().String())
+				}
+			}
+		}
+	}
+	if c.ExecuteOffChainConfig != nil {
+		if err := c.ExecuteOffChainConfig.Validate(); err != nil {
+			return fmt.Errorf("invalid execute off-chain config: %w", err)
+		}
+		for _, observer := range c.ExecuteOffChainConfig.TokenDataObservers {
+			usdcConfig := observer.USDCCCTPObserverConfig
+			if usdcConfig != nil {
```

Comment:
To future-proof this a bit more lets have a switch statement that checks the `Type` and if its not one of the supported types then we error, so that we can catch this and update the validation accordingly.

```golang
switch ty := observerConfig.Type {
  case pluginconfig.USDCCCTPHandlerType:
    validateUSDCConfig(...)
  default:
    return fmt.Errorf("unknown token observer config type: %s", ty)
}
```